lib/client.js

```diff
@@ -43,7 +46,8 @@ BaseClient.prototype.request = function(options, data, fn) {
 			"content-type": "application/json",
 			"user-agent": userAgent,
 			"content-length": typeof data === "string" ? data.length : 0
-		}
+		},
+		agent: agent
```

Comment:
a nicer way to write this could be :

`agent: this.proxy ? new HttpsProxyAgent(this.proxy) : null`

what do you think @scottgonzalez ?